host: Add import order linting

backspace opened the pull request:
This is patterned off #136 though I added significant configuration to provide a rough grouping:

![bm3 2023-09-20 18-47-03](https://github.com/cardstack/boxel/assets/43280/8f7b8867-f24a-4519-9efe-f8b35574e6b1)

From `code-mode` which has a lot more imports you can see the structure of “internal” imports:

<img width="715" alt="bm3 2023-09-20 18-49-53" src="https://github.com/cardstack/boxel/assets/43280/484c8099-8c5a-47fb-b255-530c8913cb87">

This features the results of running `lint:js:fix`.

backspace committed:
Add import/{recommended,typescript} plugins

backspace committed:
Add example files with autofix

backspace committed:
Fix grouping

lukemelia reviewed (commented):
Config is OK by me. As long as it is fix-able, I'm good with it! :)

backspace committed:
Add missed space-removal

It lets you insert arbitrary spaces so this wasn’t removed
when I collapsed the group definitions.

backspace committed:
Fix pattern-matching for card URLs

backspace committed:
Change enforcement to not apply to blueprint files

backspace committed:
Add exclusion for specific blueprint files

These are not where import insertion time-wasting is
happening.

backspace committed:
Merge branch 'main' into host/import-order-cs-6047

# Conflicts:
#	packages/host/app/components/operator-mode/code-mode.gts

backspace committed:
Add autofixes

backspace committed:
Remove import/no-duplicates

backspace committed:
Add other fixes

Why didn’t autofix do this in the CLI but I could just save
the files in the editor and that did it?

backspace committed:
Merge branch 'main' into host/import-order-cs-6047

# Conflicts:
#	packages/host/app/components/operator-mode/code-mode.gts
#	packages/host/app/components/operator-mode/detail-panel.gts
#	packages/host/app/resources/file.ts
#	packages/host/app/resources/last-modified-date.ts
#	packages/host/tests/acceptance/code-mode-test.ts
#	packages/host/tests/integration/components/card-basics-test.gts

backspace committed:
Fix autofix errors

backspace committed:
Add more lint fixes

backspace committed:
Add empty commit

backspace committed:
Merge branch 'main' into host/import-order-cs-6047

# Conflicts:
#	packages/host/tests/acceptance/operator-mode-test.ts

backspace committed:
Merge branch 'main' into host/import-order-cs-6047
